### PR TITLE
Add test for failed image generation ttthat was crashing bLasso

### PR DESCRIPTION
**Pull request by areeda**
Lasso was consistently failing with
 ```warnings.warn('Error saving {0}: {1}'.format(pngfile, str(e)))<br>
Traceback (most recent call last):<br>
  File "/home/ldvw/pyenvs/gwpy/bin/gwdetchar-lasso-correlation", line 711, in <module><br>
    page.add(htmlio.fancybox_img(img))<br>
  File "/home/ldvw/pyenvs/gwpy/lib/python2.7/site-packages/gwdetchar/io/html.py", line 748, in fancybox_img<br>
    channel = '%s-%s' % tuple(substrings[:2])<br>
TypeError: not enough arguments for format string<br>```

An example command line is:
``` /home/ldvw/pyenvs/gwpy/bin/gwdetchar-lasso-correlation 1240549218 1240556418 --channel-file /home/ldvw/public_html/lasso/areeda/H1/2019-04-29-05.00.42-7140-06/results/chanList.txt --ifo H1 --output-dir /home/ldvw/public_html/lasso/areeda/H1/2019-04-29-05.00.42-7140-06/results --primary-channel H1:GDS-CALIB_STRAIN --alpha 0.20 --remove-outliers 3.50 --cluster-coefficient 0.95 --threshold 0.00 --nproc 1 --nproc-plot 1 --trend-type minute --band-pass 46.40 51.00```
The STDOUT & STDERR are available at https://ldas-jobs.ligo.caltech.edu/~ldvw/lasso/areeda/H1/2019-04-29-05.00.42-7140-06/job/lasso.err

With this change it runs and the results are at https://ldas-jobs.ligo.caltech.edu/~ldvw/lasso/areeda/H1/2019-04-29-05.00.42-7140-08/results/
Look at the last channel: H1:ALS-C_DIFF_A_LF_OUT_DQ.mean it seems as though the remove flat channels may have missed this one.

### CODE REVIEW

File: gwdetchar/io/html.py [hunk 1]
```diff
@@ -743,22 +743,27 @@ def fancybox_img(img, linkparams=dict(), **params):
...
+            imgparams['src'] = img
+        imgparams.update(params)
+        page.img(id_='img_%s_%s' % (channel, duration), **imgparams)
+        page.a.close()
     else:
-        imgparams['src'] = img
-    imgparams.update(params)
-    page.img(id_='img_%s_%s' % (channel, duration), **imgparams)
-    page.a.close()
+        page.add('fancy-box mage was requested but no file specified.')
```
Comment (duncanmmacleod) [inline gwdetchar/io/html.py]:
```suggestion
        page.add('fancy-box image was requested but no file specified.')
```

Comment (areeda) [reply gwdetchar/io/html.py]: Tanks. fixed

File: gwdetchar/io/html.py [hunk 2]
```diff
@@ -743,22 +743,27 @@ def fancybox_img(img, linkparams=dict(), **params):
...
-    duration = substrings[-1].split('.')[0]
-    page.a(href=img, id_='a_%s_%s' % (channel, duration), **aparams)
-    imgparams = {
-        'alt': os.path.basename(img),
-        'class_': 'img-responsive',
-    }
-    if img.endswith('.svg') and os.path.isfile(img.replace('.svg', '.png')):
-        imgparams['src'] = img.replace('.svg', '.png')
+    # this is to avoid problems discovered in lasso when image creation fails
+    # but caller fails to detect it
```
Comment (alurban) [inline gwdetchar/io/html.py]: @areeda, these comments (lines 746 and 747) probably aren't needed and can be deleted

**Comment by alurban** [comment]
@areeda, do you think we still need this after the main issue you were seeing with lasso was fixed?

**Pull request closed without merging**